ProcessLib/TES/TESLocalAssembler-impl.h

```diff
@@ -227,40 +238,55 @@ TESLocalAssembler<ShapeFunction_, IntegrationMethod_, GlobalDim>::
 
 template <typename ShapeFunction_, typename IntegrationMethod_,
           unsigned GlobalDim>
-std::vector<double> const& TESLocalAssembler<
-    ShapeFunction_, IntegrationMethod_,
-    GlobalDim>::getIntPtReactionRate(std::vector<double>& /*cache*/) const
+std::vector<double> const&
+TESLocalAssembler<ShapeFunction_, IntegrationMethod_, GlobalDim>::
+    getIntPtReactionRate(const double /*t*/,
+                         GlobalVector const& /*current_solution*/,
+                         NumLib::LocalToGlobalIndexMap const& /*dof_table*/,
+                         std::vector<double>& /*cache*/) const
 {
     return _d.getData().reaction_rate;
 }
 
 template <typename ShapeFunction_, typename IntegrationMethod_,
           unsigned GlobalDim>
-std::vector<double> const& TESLocalAssembler<
-    ShapeFunction_, IntegrationMethod_,
-    GlobalDim>::getIntPtDarcyVelocityX(std::vector<double>& /*cache*/) const
+std::vector<double> const&
+TESLocalAssembler<ShapeFunction_, IntegrationMethod_, GlobalDim>::
+    getIntPtDarcyVelocity(const double /*t*/,
+                          GlobalVector const& current_solution,
+                          NumLib::LocalToGlobalIndexMap const& dof_table,
+                          std::vector<double>& cache) const
 {
-    return _d.getData().velocity[0];
-}
+    // auto const num_nodes = ShapeFunction_::NPOINTS;
+    auto const num_intpts = _shape_matrices.size();
 
-template <typename ShapeFunction_, typename IntegrationMethod_,
-          unsigned GlobalDim>
-std::vector<double> const& TESLocalAssembler<
-    ShapeFunction_, IntegrationMethod_,
-    GlobalDim>::getIntPtDarcyVelocityY(std::vector<double>& /*cache*/) const
-{
-    assert(_d.getData().velocity.size() > 1);
-    return _d.getData().velocity[1];
-}
+    auto const indices = NumLib::getIndices(_element.getID(), dof_table);
```

Comment:
Yes, probably. But in other processes the pattern is the same, cf. HT. As far as I can see.